File: leetcode-python-go/2347. 最好的扑克手牌/main.py
from typing import List
from collections import Counter

ranks = [4,4,2,4,4]
suits = ["d","a","a","b","c"]

def bestHand(ranks: List[int], suits: List[str]) -> str:
    if len(set(suits)) == 1:
        return "Flush"
    h = Counter(ranks)
    if len(h) == 5:
        return "High Card"
    for [a, b] in h.items():
        if b > 2:
            return "Three of a Kind"
    return "Pair"

print(bestHand(ranks, suits))

File: leetcode-python-go/2347. 最好的扑克手牌/main.go
package main

import (
	"bytes"
	"fmt"
)

func bestHand(ranks []int, suits []byte) string {
	if bytes.Count(suits, suits[:1]) == 5 {
		return "Flush"
	}
	cnt, pair := map[int]int{}, false
	for _, r := range ranks {
		cnt[r]++
		if cnt[r] == 3 {
			return "Three of a Kind"
		}
		if cnt[r] == 2 {
			pair = true
		}
	}
	if pair {
		return "Pair"
	}
	return "High Card"
}

func main() {
	ranks := []int{4, 4, 2, 4, 4}
	suits := []byte{byte("d"[0]), byte("a"[0]), byte("a"[0]), byte("b"[0]), byte("c"[0])}
	results := bestHand(ranks, suits)
	fmt.Println(results)
}
